spec/models/publish_services_event_metadata_spec.rb

```diff
@@ -51,4 +53,87 @@
       expect(event).not_to be_persisted
     end
   end
+  context 'polling' do
+    let!(:service) { create(:service) }
+    let(:hub_response_for_signing_certificate) {
+      [{
+        issuerId: service.entity_id,
+        certificate: sp_signing_certificate.value,
+        keyUse: 'Signing',
+        federationEntityType: 'RP',
+      }].to_json
+    }
+
+    after(:each) do
+      SpComponent.destroy_all
+      MsaComponent.destroy_all
+    end
+
+    context 'does not occur' do
+      require 'polling/dev_cert_status_updater'
+      let(:sp_signing_certificate) { create(:sp_signing_certificate) }
+      it 'updates certificate in_use_at' do
+
+        expect(Certificate.find_by_id(sp_signing_certificate.id).in_use_at).to be_nil
+        dev_cert_status_updater = DevCertStatusUpdater.new
+        allow(subject).to receive(:poll).with(scheduler: Polling::Scheduler.new, status_updater: dev_cert_status_updater, hub_api: HUB_CONFIG_API)
+        allow(dev_cert_status_updater).to receive(:update_hub_usage_status_for_cert).with(HUB_CONFIG_API, sp_signing_certificate)
+          .and_return(CertificateInUseEvent.create(certificate: sp_signing_certificate))
+
+        create(:assign_sp_component_to_service_event, service: service, sp_component_id: sp_signing_certificate.component.id)
+        expect(Certificate.find_by_id(sp_signing_certificate.id).in_use_at).not_to be_nil
+      end
+    end
+
+    context 'occurs' do
+      let(:msa_encryption_certificate) { create(:msa_encryption_certificate) }
```

Comment:
this is not in use, but I think it should be. We should be testing for both encryption and signing certs since there's a slight difference

Reply:
Thanks, I agree very much with this; if I had I would have probably found the issue with your comment on `enabled : true` above